src/planning.py

```diff
@@ -417,6 +418,79 @@ def _run_low_level_search(task: Task, option_model: _OptionModelBase,
     return [], True
 
 
+def _run_low_level_plan(
+        task: Task, option_model: _OptionModelBase, plan: List[_Option],
+        timeout: float, max_horizon: int
+) -> Tuple[LowLevelTrajectory, bool]:  # pragma: no cover
+    """Runs a plan on an option model to generate a low level trajectory.
+
+    Returns a LowLevelTrajectory and a boolean. If the boolean is True,
+    the option sequence successfully executed to achieve the goal and
+    generated a LowLevelTrajectory. Otherwise, it returns and empty list
+    and False.
+    """
+    start_time = time.time()
+    cur_idx = 0
+    # The number of actions taken by each option in the plan. This is to
+    # make sure that we do not exceed the task horizon.
+    num_actions_per_option = [0 for _ in plan]
+    traj: List[State] = [task.init] + [DefaultState for _ in plan]
+    actions: List[Action] = [Action(np.array([0.0])) for _ in plan]
+    while cur_idx < len(plan):
+        if time.time() - start_time > timeout:
+            return LowLevelTrajectory([], [], False), False
+        state = traj[cur_idx]
+        option = plan[cur_idx]
+        cur_idx += 1
+        if option.initiable(state):
+            try:
+                next_state, num_actions = \
+                    option_model.get_next_state_and_num_actions(state, option)
```

Comment:
is this function deterministic? if so, i don't see why we need to do backtracking in this function, because what would change between subsequent iterations?